src/app/reporting/Engine.cpp

```diff
@@ -138,12 +138,15 @@ CHIP_ERROR Engine::BuildSingleReportDataAttributeReportIBs(ReportDataMessage::Bu
                     attributeReportIBs.Rollback(attributeBackup);
                     apReadHandler->SetAttributeEncodeState(AttributeValueEncoder::AttributeEncodeState());
 
-                    // Try to encode our error as a status response.
-                    err = attributeReportIBs.EncodeAttributeStatus(pathForRetrieval, StatusIB(err));
-                    if (err != CHIP_NO_ERROR)
+                    if (err != CHIP_ERROR_NO_MEMORY)
```

Comment:
Why?  Just because something failed to allocate memory somewhere encoding a (possibly large) data response doesn't mean we can't try to encode a status response.

Reply:
Ah, I see.  This is an attempt to chunk on the level of the whole attribute.

It seems to me that the behavior should then be as follows:

1. If error is not CHIP_ERROR_BUFFER_TOO_SMALL and is not CHIP_ERROR_NO_MEMORY, rollback and encode status.
2. Otherwise, if partial data allowed, save the encode state.
3. Otherwise roll back.  If we have already encoded some chunks, we are done; otherwise encode status.

Thoughts?

Reply:
yep, agree, updated, thanks